Shopify/cli
# Don't transform configurations into camelCase

### WHY are these changes introduced?
We had some inconsistencies in how we loaded configuration toml files (both for app and extensions). We were transforming the root keys of the config file to camelCase, but not any nested key which didn't make sense and was causing issues with some new projects.



### WHAT is this pull request doing?
- Remove the camelCase transformation when loading config files
- Update all the Schemas to expect `snake_case`
- Update all tests


### How to test your changes?
- Everything still works... my recommendation would be to create an app on main with some extensions and custom config, print that config when calling `info` and do the same from this branch.
- Also, dev and deploy should still work


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
